server/homebrew.api.spec.js

```diff
@@ -0,0 +1,377 @@
+				params : {
+					id : '123456789012abcdefghijkl'
+				}
+			});
+
+			expect(id).toEqual('abcdefghijkl');
+			expect(googleId).toEqual('123456789012');
+		});
+	});
+
+	describe('getBrew', ()=>{
+		const toBrewPromise = (brew)=>new Promise((res)=>res({ toObject: ()=>brew }));
+		const notFoundError = 'Brew not found in Homebrewery database or Google Drive';
+
+		it('returns middleware', ()=>{
+			const getFn = api.getBrew('share');
+			expect(getFn).toBeInstanceOf(Function);
+		});
+
+		it('should fetch from mongoose', async ()=>{
+			const testBrew = { title: 'test brew', authors: [] };
+			api.getId = jest.fn(()=>({ id: '1', googleId: undefined }));
+			model.get = jest.fn(()=>toBrewPromise(testBrew));
+
+			const fn = api.getBrew('share', true);
+			const req = { brew: {} };
+			const next = jest.fn();
+			await fn(req, null, next);
+
+			expect(req.brew).toEqual(testBrew);
+			expect(next).toHaveBeenCalled();
+			expect(api.getId).toHaveBeenCalledWith(req);
+			expect(model.get).toHaveBeenCalledWith({ shareId: '1' });
+		});
+
+		it('should handle mongoose error', async ()=>{
+			api.getId = jest.fn(()=>({ id: '1', googleId: undefined }));
+			model.get = jest.fn(()=>new Promise((_, rej)=>rej('Unable to find brew')));
+
+			const fn = api.getBrew('share', true);
+			const req = { brew: {} };
+			const next = jest.fn();
+			let err;
+			try {
+				await fn(req, null, next);
+			} catch (e) {
+				err = e;
+			}
+
+			expect(err).toEqual(notFoundError);
+			expect(req.brew).toEqual({});
+			expect(next).not.toHaveBeenCalled();
+			expect(api.getId).toHaveBeenCalledWith(req);
+			expect(model.get).toHaveBeenCalledWith({ shareId: '1' });
+		});
+
+		it('changes tags from string to array', async ()=>{
+			const testBrew = { title: 'test brew', authors: [], tags: 'tag' };
+			api.getId = jest.fn(()=>({ id: '1', googleId: undefined }));
+			model.get = jest.fn(()=>toBrewPromise(testBrew));
+
+			const fn = api.getBrew('share', true);
+			const req = { brew: {} };
+			const next = jest.fn();
+			await fn(req, null, next);
+
+			expect(req.brew.tags).toEqual([]);
+			expect(next).toHaveBeenCalled();
+		});
+
+		it('throws if invalid author', async ()=>{
+			api.getId = jest.fn(()=>({ id: '1', googleId: undefined }));
+			model.get = jest.fn(()=>toBrewPromise({ title: 'test brew', authors: ['a'] }));
+
+			const fn = api.getBrew('edit', true);
+			const req = { brew: {} };
+
+			let err;
+			try {
+				await fn(req, null, null);
+			} catch (e) {
+				err = e;
+			}
+
+			expect(err).toEqual('Current logged in user does not have access to this brew.');
+		});
+
+		it('does not throw if no authors', async ()=>{
+			api.getId = jest.fn(()=>({ id: '1', googleId: undefined }));
+			model.get = jest.fn(()=>toBrewPromise({ title: 'test brew', authors: [] }));
+
+			const fn = api.getBrew('edit', true);
+			const req = { brew: {} };
+			const next = jest.fn();
+			await fn(req, null, next);
+
+			expect(next).toHaveBeenCalled();
```

Comment:
Do we expect `{ title: 'test brew', authors: [] }` to be returned or we expect to see an empty brew? I would suggest that we test `req.brew` result as well: for purposes of documenting API behavior and testing it.